### PR TITLE
perf: set compat_level when calling to_arrow

### PR DESCRIPTION
**Pull request by ruihe774**
This depends on https://github.com/pola-rs/polars/pull/17371.

In my naive benchmark, with large string columns, it improves the speed to ~10x.

### CODE REVIEW

**Comment by ruihe774** [comment]
I'm still wondering how to ensure forward compatibility with `future=True`. When a old version of `pyo3-polars` is used together with a newer version of python polars which has introduced incompatible arrow data structure updates, it can be broken. One solution is to check the version of polars before setting `future=True`, and another solution is to introduce versioning like `future=1`.

As for backward compatibility, we can simply retry with `future=True` unset if it fails.

**Comment by ruihe774** [comment]
https://github.com/pola-rs/polars/pull/17421

**Comment by ruihe774** [comment]
> [pola-rs/polars#17421](https://github.com/pola-rs/polars/pull/17421)

I've made this PR work with it. In `PySeries::into_py` and `PySeries::extract_bound`, newest compatibility level supported by both sides is negotiated and selected automatically. In this way, both backward compatibility and forward compatibility are achieved.

File: Cargo.toml [hunk 1]
```diff
@@ -13,3 +13,10 @@ polars-core = { version = "0.41.0", default-features = false }
 polars-ffi = { version = "0.41.0", default-features = false }
 polars-plan = { version = "0.41.0", default-feautres = false }
 polars-lazy = { version = "0.41.0", default-features = false }
+
+[patch.crates-io]
```
Comment (ritchie46) [inline Cargo.toml:17]: Can we point this to the main branch until `0.42` is released.

Comment (ruihe774) [reply Cargo.toml:17]: Cargo only recognizes patches defined in root project and not in dependencies. `pyo3-polars` is meant to be a dependency of projects that use it. So unless they declare patches in their `Cargo.toml`, patches are not in effect.

Patches declared here is only for testing. We can only wait for `0.42`.

Comment (ritchie46) [reply Cargo.toml:17]: Alright. 0.42 is planned in ca. 2 weeks.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**